### ISSUE TITLE
Cant find public server.

### ISSUE THREAD

**Issue by dipievil** [lloesche/valheim-server-docker#225]
Even marking SERVER_PUBLIC=true cant find my server, but everyone can connect. 
![server listing](https://i.imgur.com/4N5mEjo.png)

![portainer stack configuration](https://i.imgur.com/Z6YGsrt.png)

When connecting the server the name is weird.
![ingame](https://i.imgur.com/8jWO9r3.png)

What else do I need to change/check to make it public?

**Comment by lloesche** [comment]
If you have opened ports 2456 and 2457 UDP and they are publicly accessible then there is not much more you can do. This is a widespread known issue that has been asked here a couple of times. You'll even find that sometimes your friends from other continents will be able to see the server while others won't.
The most reliable way is to [bookmark the server](https://github.com/lloesche/valheim-server-docker#steam-server-favorites--lan-play).

Visibility aside, check out our [docker-compose.yaml](https://github.com/lloesche/valheim-server-docker/blob/main/docker-compose.yaml). You are missing some options that might make it so your world won't be saved on time when you stop the container.

**Comment by dipievil** [comment]
Do you have any ideas about the server name has the weird name ingame? Maybe something related to the publicity problem?

**Comment by lloesche** [comment]
@dipievil for me restarting the server does not help. I can never see my server in-game and I have no NAT or anything like that. I always connect using Steam Server favorites.

**Comment by greyltc** [comment]
I wonder if this has to do with the GAME field somehow not being populated correctly for the public server advertisement. I've noticed that after I follow the readme instructions to add my server to my steam favorites list, everything shows there as I expect except the GAME name is blank (and I expect if it was working correctly, it should show `Valheim`). I can see some servers actually show the game correctly, but most don't. That's strange.

There are a few other possibly related things I've noticed:
- The status.json file contains `"error": null` is `null` the expected error value here?
- The readme has a `STATUS_HTTP_PORT` defaulting to 80. Does that need to be exposed/forwarded by docker for this to work? I don't see that being done anywhere.

**Comment by dipievil** [comment]
In my case my server even doesnt show up on the Steam Server List but I can connect and play.

`"error": null` means that the server has no errors at all.

If you can see the status.json means that the port in STATUS_HTTP_PORT is already exposed.

**Comment by InB4DevOps** [comment]
If you can not find your server then the server is (most likely) too far aways from you (geographically speaking).

I had similar issues with my own server. A friend from Switzerland could never see it in the list. Once connected to a nearby VPN server the server appeared in his server list. 

TL;DR: there is nothing you can do to fix this problem.

**Comment by greyltc** [comment]
> "error": null means that the server has no errors at all.

@dipievil Cool, thanks! Thought it was worth checking just because if I wrote the underlying code, `"error": null` would mean I forgot to define the error variable or similar. (whereas `"error": 0` might mean no error)

>  If you can see the status.json means that the port in STATUS_HTTP_PORT is already exposed.

I exposed it myself just so I could inspect status.json for errors so that I might be able to get to the reason why my server is invisible in the public lists.

@InB4DevOps 
Wait, so you're saying somebody (Valve, I guess?) is putting a secret geographical filter on the server list that we can't control!?
That explains perfectly what I'm seeing though. Bummer they've made the decision that my European friends and I should have a bit more trouble connecting to my server which happens to be in California!

Even though the underlying reason for this is likely explained now, I'll write down another silly question here that might be related but  probably is not and also is probably not even one that makes sense to post here: 
Why does [this similar container](https://github.com/mbround18/valheim-docker) need three ports mapped to run (PORT through PORT+2) whereas this one only needs two (PORT and PORT+1)?

**Comment by InB4DevOps** [comment]
@greyltc I don't think that it's a secret filter. It's just a mechanic to limit the number of servers the game can query from Steam. 
Just look at the number of pulls for this container alone. There might be tens of thousands of servers in the world at the moment. How long would you want to wait to have them all pinged? :)

What we'd need here is a reqion filter so that you could query servers from different regions in the world.

**Comment by greyltc** [comment]
Yeah, reducing the scope of the servers you ping would limit a bunch of icmp flooding-type traffic I guess. But they could still show all servers in the list and then only ping your regional ones or the ones you click on.

**Comment by lloesche** [comment]
> * The readme has a `STATUS_HTTP_PORT` defaulting to 80. Does that need to be exposed/forwarded by docker for this to work? I don't see that being done anywhere.

`STATUS_HTTP` defaults to `false` so for most users it won't be active. The idea is, if you are interested enough to turn on the status http server and parse its JSON output you also know how to expose its port. Which you did. So all is good 🙂

**Comment by lloesche** [comment]
> @dipievil Cool, thanks! Thought it was worth checking just because if I wrote the underlying code, `"error": null` would mean I forgot to define the error variable or similar. (whereas `"error": 0` might mean no error)

I can see that. In the Python code it reads `"error": None` which I feel is "more correct" than `"error": int(0)`. During marshalling the `None` then turns into `null` via the JSON exporter.
